modules/dnn/test/test_layers.cpp

```diff
@@ -777,6 +777,97 @@ TEST(Layer_PriorBox, squares)
     normAssert(out.reshape(1, 4), target);
 }
 
+TEST(Layer_Test_DWconv_Prelu, Accuracy)
+{
+    // Test case
+    // input       img size 3x16x16  value all 1
+    //   |
+    //   v
+    // dw_conv     weight[0]=-1 weight[0]=-2 weight[0]=-3 no_bias
+    //   |
+    //   v
+    // prelu       weight={1,2,3}
+    //   |
+    //   v
+    // output      out size 3x14x14  if right: out[0]=-9 out[0]=-36 out[0]=-81
+    //             but current opencv output: out[0]=-27 out[0]=-54 out[0]=-81
+
+    Net net;
+    //layer 1: dwconv
+    LayerParams lp;
+    lp.name = "dwconv";
+    lp.type = "Convolution";
+    lp.set("kernel_size", 3);
+    lp.set("num_output", 3);
+    lp.set("pad", 0);
+    lp.set("group", 3);
+    lp.set("stride", 1);
+    lp.set("engine", "CAFFE");
+    lp.set("bias_term", "false");
+
+    std::vector<int> weightsShape(4);
+    weightsShape[0] = 3;  // #outChannels
+    weightsShape[1] = 1;  // #inpChannels / group
+    weightsShape[2] = 3;  // height
+    weightsShape[3] = 3;  // width
+    Mat weights(weightsShape, CV_32F, Scalar(1));
+
+    //assign weights
+    for (int i = 0; i < weightsShape[0]; ++i)
+    {
+        for (int j = 0; j < weightsShape[1]; ++j)
+        {
+            for (int k = 0; k < weightsShape[2]; ++k)
+            {
+                for (int l = 0; l < weightsShape[3]; ++l)
+                {
+                    weights.ptr<float>(i, j, k)[l]=-1*(i+1);
+                }
+            }
+        }
+    }
+    lp.blobs.push_back(weights);
+    net.addLayerToPrev(lp.name, lp.type, lp);
+
+    //layer 2: prelu
+    LayerParams lpr;
+    lpr.name = "dw_relu";
+    lpr.type = "PReLU";
+    std::vector<int> weightsShapep(2);
+    weightsShapep[0] = 1;  // #outChannels
+    weightsShapep[1] = 3;  // #inpChannels / group
+    Mat weightsp(weightsShapep, CV_32F, Scalar(1));
```

Comment:
Just create a `cv::Mat` with a single row and number of columns equals to number of output channels (3):
```cpp
Mat weightsp(1, 3, CV_32F, Scalar(1));
```
It's better to replace number of output channels to a special constant at the beginning of the test. Use it in `lp.set("num_output", 3);` too.